### ISSUE TITLE
Certificate signed error displayed on installing agent under user-defined fleet-server

### ISSUE THREAD

**Issue by dikshachauhan-qasource** [elastic/beats#26884]
**Kibana version:** 7.13.4 snapshot Kibana Cloud-Prod environment

**Host OS and Browser version:** Chrome, Firefox and edge

**Preconditions:**
1. 7.13.4 snapshot Kibana Cloud-Prod environment should be available.

**Build Details:**
```
BUILD 41022
COMMIT 024b8904d1508252df7cb41ac98f48c48f7bcb33
Artifact link: https://staging.elastic.co/7.13.4-294dae98/summary-7.13.4.html

```
2. A policy having Fleet server integration should be available.
3. Install a fleet-server agent using above policy on windows.
4. Create a inbound rule for 8220 port in firewall settings on windows machine.
5. Update fleet-server ip details in fleet settings with input: https://10.0.x.x:8220

**Steps to reproduce:**
	- Login to Kibana.
	- Navigate to Agents tab.
	- Observe, user fleet-server agent is running properly, installed on a windows machine.
	- Create a new user policy having system integration.
	- On linux or mac os [with sip_enabled], attempt to install a new agent using above fleet server host url.
	- Observe, certificate signed by unknown authority error displays.

**Actual result:**
- Unable to install agent on user-created fleet-servers.

**Expected result:**
Should be able to install agents on user-created fleet-servers.

**Screenshot:**
**Fleet-server details:**
![image](https://user-images.githubusercontent.com/72375798/125611098-55d3407b-09a2-4a73-aa4f-b533e4d31d46.png)

**Fleet-settings:**
![image](https://user-images.githubusercontent.com/72375798/125611405-575a5afc-9b08-496e-8c6b-fb532d7530b6.png)

**Error on mac os:**
![image](https://user-images.githubusercontent.com/72375798/125611661-029e7363-40f9-4a48-aba6-d45864bc9ddf.png)

**No Errors displayed when installed agent directing using default fleet-server:**
![image](https://user-images.githubusercontent.com/72375798/125611925-79b523c3-8b57-48e3-bb5f-f44771662216.png)

**Comment by dikshachauhan-qasource** [comment]
@manishgupta-qasource Please review

**Comment by manishgupta-qasource** [comment]
Reviewed & Assigned to @andresrc 

CC: @EricDavisX

**Comment by blakerouse** [comment]
This is correct behavior. That is because you have 1 Fleet Server that is being ran by Cloud and it has a valid (non self-signed) SSL certificate and then you have another Fleet Server that ran on the Windows machine that is using a self-signed certificate.

Because 1 of the Fleet Servers has a self-signed certificate you must install the other Elastic Agent with `--insecure`.

If you could get a valid DNS name with a valid certificate for the Windows machine then this would work without the `--insecure` flag.

**Comment by dikshachauhan-qasource** [comment]
Hi @EricDavisX 

Further, we check out agent logs when installed agent directing using default fleet-server. And on scrolling, we found error logs related to certificates in agent logs too.

Agent logs:
[agent_logs_for mac_OS_when_installed_using_default_fleet_server.txt](https://github.com/elastic/beats/files/6815648/agent_logs_for.mac_OS_when_installed_using_default_fleet_server.txt)
[errors_logs_on_linuxagent_with_default_fleet_server.txt](https://github.com/elastic/beats/files/6815649/errors_logs_on_linuxagent_with_default_fleet_server.txt)


Logs for user-defined fleet-server: 
[user-created-fleet-server-logs.txt](https://github.com/elastic/beats/files/6815653/user-created-fleet-server-logs.txt)

Please let us know if anything else is required.

Thanks
QAS

**Comment by EricDavisX** [comment]
@dikshachauhan-qasource thanks for raising it.  Indeed until the team decides to support the mixed "hybrid" environment a user who wants to use cloud for the ELK stack will want to disable the Fleet / APM server or remove it from the Fleet Settings at least.
